charts/jaeger/templates/_helpers.tpl

```diff
@@ -268,7 +268,11 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 */}}
 {{- define "elasticsearch.client.url" -}}
 {{- $port := .Values.storage.elasticsearch.port | toString -}}
-{{- printf "%s://%s:%s" .Values.storage.elasticsearch.scheme .Values.storage.elasticsearch.host $port }}
+{{- $host := .Values.storage.elasticsearch.host }}
+{{- if .Values.provisionDataStore.elasticsearch }}
+{{- $host = printf "%s-elasticsearch" .Release.Name }}
+{{- end }}
```

Comment:
this looks like it overrides the user-provided host. Shouldn't it be more like

```suggestion
{{- $host := printf "%s-elasticsearch" .Release.Name }}
{{- if .Values.storage.elasticsearch.host}}
  {{- $host = .Values.storage.elasticsearch.host }}
{{- end }}
```

Reply:
there is always a default value present in `values.yaml `  so it will always override the `.Values.provisionDataStore.elasticsearch`
and additionally it doesn't matter what you pass to `.Values.storage.elasticsearch.host` since in provision storage it is gonna use it's own chart so changing host name will not affect the name of provision es service

Reply:
it doesn't make sense to have a `host` property in values.yaml if it has no impact. Should we set it to null then? Because in case the user manages their own Elastic cluster they may still need to provide a different name.